File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/options/RemovePolylinesByIdsOptions.java
package io.capawesome.capacitorjs.plugins.maplibre.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.maplibre.MapLibreHelper;
import io.capawesome.capacitorjs.plugins.maplibre.classes.CustomExceptions;
import java.util.List;

public class RemovePolylinesByIdsOptions {

    @NonNull
    private final String mapId;

    @NonNull
    private final List<String> polylineIds;

    public RemovePolylinesByIdsOptions(@NonNull PluginCall call) throws Exception {
        JSArray polylineIdsArray = call.getArray("polylineIds", null);
        if (polylineIdsArray == null) {
            throw CustomExceptions.POLYLINE_IDS_MISSING;
        }
        this.mapId = MapLibreHelper.getMapIdFromCall(call);
        this.polylineIds = MapLibreHelper.createStringList(polylineIdsArray);
    }

    @NonNull
    public String getMapId() {
        return mapId;
    }

    @NonNull
    public List<String> getPolylineIds() {
        return polylineIds;
    }
}

File: packages/maplibre/ios/Plugin/Classes/Options/RemovePolylinesByIdsOptions.swift
import Capacitor
import Foundation

@objc public class RemovePolylinesByIdsOptions: NSObject {
    let mapId: String
    let polylineIds: [String]

    init(_ call: CAPPluginCall) throws {
        self.mapId = try MapLibreHelper.getString(call, "mapId", .mapIdMissing)
        self.polylineIds = try MapLibreHelper.getStringArray(call, "polylineIds", .polylineIdsMissing)
    }
}
